lms/djangoapps/support/static/support/jsx/program_enrollments/inspector.jsx

```diff
@@ -2,55 +2,174 @@ import React from 'react';
+    <hr />
+  </div>
+);
+
+
+const renderVerificationSection = verificationStatus => (
+  <div>
+    <h3>
+      ID Verification
+    </h3>
+    <div><span>Status: </span> {verificationStatus.status}</div>
+    <div><span>Verification Error: </span> {verificationStatus.error}</div>
+    <div><span>Should Display Verification: </span> {verificationStatus.should_display}</div>
+    <div><span>Verification Expiration Date: </span>
+      {verificationStatus.verification_expiry}
     </div>
-    <Button label="Search" type="submit" className={['btn', 'btn-primary']} />
-  </form>
+    <hr />
+  </div>
+);
+
+const renderEnrollmentsSection = enrollments => (
+  <div>
+    <h3>
+      Program Enrollments
+    </h3>
+    {enrollments.map(enrollment => (
+      <div key={enrollment.program_uuid}>
+        <div>Program {enrollment.program_uuid} Enrollments</div>
+        <div> <span>Status: </span> {enrollment.status} </div>
+        <div> <span>Created: </span> {enrollment.created} </div>
+        <div> <span>Last updated: </span> {enrollment.modified} </div>
+        <div> <span>External User Key: </span> {enrollment.external_user_key} </div>
+        {enrollment.program_course_enrollments && enrollment.program_course_enrollments.map(
+          programCourseEnrollment => (
+            <div key={programCourseEnrollment.course_key}>
+              <br />
+              <h4>Course {programCourseEnrollment.course_key}</h4>
+              <div> <span>Status: </span> {programCourseEnrollment.status} </div>
+              <div> <span>Created: </span> {programCourseEnrollment.created} </div>
+              <div> <span>Last updated: </span> {programCourseEnrollment.modified} </div>
+              {programCourseEnrollment.course_enrollment && (
+                <div>
+                  <br />
+                  <h4>Linked course enrollment</h4>
+                  <div><span>Course ID: </span>
+                    {programCourseEnrollment.course_enrollment.course_id}
+                  </div>
+                  <div> <span>Is Active: </span>
+                    {programCourseEnrollment.course_enrollment.is_active}
+                  </div>
+                  <div> <span>Mode / Track: </span>
+                    {programCourseEnrollment.course_enrollment.mode}
+                  </div>
+                </div>
+              )}
+            </div>
+          ))}
+      </div>
+    ))}
+    <hr />
+  </div>
+);
+
+
+export const ProgramEnrollmentsInspectorPage = props => (
+  <div>
+    {JSON.stringify(props.learnerInfo) !== '{}' && (<h2> Search Results </h2>)}
+    {props.learnerInfo.user &&
+      renderUserSection(props.learnerInfo.user)}
+    {props.learnerInfo.id_verification &&
+      renderVerificationSection(props.learnerInfo.id_verification)}
+    {props.learnerInfo.enrollments &&
+      renderEnrollmentsSection(props.learnerInfo.enrollments)}
+    <form method="get">
+      <h2>Search For A Masters Learner Below</h2>
```

Comment:
When I load this page, it's not clear to me which fields are required. Do I have to fill in all three? Can I provide just one of them? Which combinations are valid?

Reply:
This is a place where a Paragon [FieldSet](https://edx.github.io/paragon/components/fieldset) component would be very useful; it allows you to display validation errors before and after a user clicks the submit button. But I know it's not available in the version of Paragon used in edx-platform. That's something I'd add to the follow up UI improvement ticket.

At a minimum, I'd add `(required)` next to any required fields here. Better yet, you could include a string describing what values need to be supplied and in what combinations and potentially even disable the Submit button if those conditions aren't met.

Reply:
As I test this more, I see that if I supply only an external user key, the page just reloads with no change and doesn't show me an error. If it's not valid to supply only an external user key without an organization as well, I would either return an error in your get handler in `program_enrollments.py` or make it clear that you need to supply both in the UI. I had to dig into the code to figure out that I had to supply both the organization and the external user key. It makes sense why that isn't a valid combination of fields, but I think not letting the user know what went wrong will confuse them.